### PR TITLE
Common play skill tests

### LINKED ISSUE #2332: Add basic testcases for the audioservice interface
Add tests for the [audioservice](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/skills/audioservice.py) interface for skills.

Basically the tests need to create a mock bus and create the audioservice with

```python
bus_mock = Mock()`
as = AudioService(bus_mock)
```

Then do calls against the objects methods and ensure that the expected messages are sent on the mock_bus.

If there are questions just comment here or join [our chat](https://chat.mycroft.ai)!

### PR DESCRIPTION
**Pull request by forslund**
## Description
This adds unittests for the common play framework along with the semi-replated audioservice interface. Resolves #2332 

There are a couple of minor fixes and improvements
- track_info now uses the wait_for_response mechanic
- the queue now accepts tracks with mime info.
- A bunch of docstrings are improved

## How to test
Ensure that the travis tests passes.

## Contributor license agreement signed?
CLA [ Yes]

### CODE REVIEW

File: test/unittests/skills/test_audioservice.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        self.assertEqual(message.data['tracks'], ['file:///hello_nasty.mp3'])
+        self.assertEqual(message.data['repeat'], True)
+        self.assertEqual(message.data['utterance'], 'on vlc')
+
+    def test_queue(self):
+        bus = mock.Mock(name='bus')
+        audioservice = AudioService(bus)
+
+        # Test proper uri
+        audioservice.queue('file:///hello_nasty.mp3')
```
Comment (chrisveilleux) [inline test/unittests/skills/test_audioservice.py]: lots of duplicate code here. consider a function where you pass an argument for the queue and one for the actual value in the second assert statement

File: test/unittests/skills/test_common_play_skill.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    Useful for assert_called_with arguments.
+    """
+    def __eq__(self, other):
+        return callable(other)
+
+
+class TestCommonPlay(unittest.TestCase):
+    def test_lifecycle(self):
+        skill = CPSTest()
+        bus = mock.Mock(name='bus')
```
Comment (chrisveilleux) [inline test/unittests/skills/test_common_play_skill.py]: Use a setup method here that executes these two lines of code.  Make skill and bus instance attributes

File: test/unittests/skills/test_audioservice.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    def test_lifecycle(self):
+        bus = mock.Mock(name='bus')
+        audioservice = AudioService(bus)
+        self.assertEqual(audioservice.bus, bus)
+
+    def test_play(self):
+        bus = mock.Mock(name='bus')
+        audioservice = AudioService(bus)
+
+        # Test proper uri
```
[the remark this answers is not in the document]
Comment (forslund) [reply test/unittests/skills/test_audioservice.py]: No, the it's just variations of input so I don't think it should be separate tests. If each check required a bunch of extra setups separating the tests would be called for but not when it's as simple as this.

File: test/unittests/skills/test_audioservice.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
+import unittest
+from unittest import mock
+
+from mycroft.messagebus import Message
+from mycroft.skills.audioservice import AudioService
+
+
+class TestAudioService(unittest.TestCase):
+    def test_lifecycle(self):
+        bus = mock.Mock(name='bus')
```
[the remark this answers is not in the document]
Comment (forslund) [reply test/unittests/skills/test_audioservice.py]: I personally want the tests to be as easy to follow as possible without jumping around to see exactly how the setup was done. If this requires some duplication of setup code that's fine by me.

[the remark this answers is not in the document]
Comment (forslund) [reply test/unittests/skills/test_audioservice.py]: Especially since one of the tests include the init I don't want it to be in the setup

**Comment by forslund** [comment]
I refactored out the setup code in the test_common_play_skill since after looking at it again it was slightly complex and it made sense to move it out. I also split up the test of match play query and moved it to a separate testcase class to make sure the test was grouped logically.

The test_audioservice setup was so trivial so I left it as is, moving it to class members just makes it more fragmented and difficult to read.

**Comment by forslund** [comment]
Did some more refactoring of the tests, the multi tests have been split out into separate classes to make sure it's shown that they're related.

Some of the code duplication has been removed while some was kept to keep the logic of the test clear.

**Pull request merged**